yb-voyager/src/queryparser/query_parser.go

```diff
@@ -41,6 +41,11 @@ func (qp *QueryParser) Parse() error {
 }
 
 func (qp *QueryParser) GetUnsupportedQueryConstructs() ([]utils.UnsupportedQueryConstruct, error) {
+	if qp.ParseTree == nil {
```

Comment:
Looks like if parse fails, we're moving on, therefore we run into this. 

I'd suggest: return error. GetUnsupportedQueryConstructs() should NOT silently return no error in such a scenario. (Anyway the caller logs the error and moves on for now)

Reply:
done